doc/source/operations/ubuntu-jammy.rst

```diff
@@ -0,0 +1,523 @@
+``stackhpc_pulp_sync_for_local_container_build`` to ``true`` before syncing
+container images.
+
+To build the overcloud images locally and push them to the local Pulp server:
+
+.. code-block:: console
+
+   kayobe overcloud container image build --push
+
+It is possible to build a specific set of images by supplying one or more
+image name regular expressions:
+
+.. code-block:: console
+
+   kayobe overcloud container image build --push ironic- nova-api
+
+Deploy the latest container images
+----------------------------------
+
+Make sure you deploy the latest containers before this upgrade:
+
+.. code-block:: console
+
+   kayobe seed service deploy
+   kayobe overcloud service deploy
+
+Common issues for all host types
+================================
+
+-  Interface names regularly change during upgrades, usually gaining the
+   ``np0`` suffix. This cannot easily be resolved. The upgrade script
+   configures networking both before and after rebooting to apply the upgrade.
+   Setting the interface statically in a kayobe-config fails during one of
+   these. This can be worked around by adding a ``sed`` command to the upgrade
+   script between the upgrade playbook step and the host configure step e.g.
+
+   .. code-block:: bash
+
+      kayobe playbook run $KAYOBE_CONFIG_PATH/ansible/ubuntu-upgrade.yml -e os_release=jammy --limit $1
+      sed -i -e 's/"ens1"/"ens1np0"/g' -e 's/"ens2"/"ens2np0"/g' $KAYOBE_CONFIG_PATH/environments/production/inventory/group_vars/compute/network-interfaces
+      kayobe overcloud host configure --limit $1 --kolla-limit $1 -e os_release=jammy
+
+   Remember to reset the change before upgrading another upgrade (or add a
+   second ``sed`` command to automate the process)
+-  Disk names can change during upgrades. This can be resolved in kayobe-config
+   once the new name is known (i.e. after the first upgrade) and applied by
+   re-running ``host configure`` for the affected host.
+-  Timeouts can become an issue with some hardware. The host will reboot once
+   or twice depending on whether it needs to apply package updates. Edit the
+   timeouts in the upgrade playbook (``ubuntu-upgrade.yml``) where required.
+
+Controllers
+===========
+
+Upgrade controllers *one by one*, ideally upgrading the host with the Virtual
+IP (VIP) last. Before upgrading a host with the VIP, restart the ``keepalived``
+container to fail it over to another controller.
+
+.. code-block:: bash
+
+   sudo docker restart keepalived
+
+Always back up the overcloud DB before starting:
+
+.. code-block:: bash
+
+   kayobe overcloud database backup
+
+Potential issues
+----------------
+
+-  In both testing and production, RabbitMQ has fallen into an error state
+   during controller upgrades. Keep an eye on the RabbitMQ Grafana dashboard and
+   if errors begin to increase, use the ``rabbitmq-reset`` playbook:
+
+   .. code-block:: bash
+
+      kayobe playbook run $KAYOBE_CONFIG_PATH/ansible/rabbitmq-reset.yml
```

Comment:
Do you have any details on what broke with Rabbit? It would be good to avoid a full reset if we can. For example, if only one or two services are complaining it can be sufficient to restart just their API containers so that new queues are created.

Reply:
Sadly no. I just saw a lot (250+) of tempest tests fail, quickly checked logs & grafana and saw some bad signs. I just ran the playbook and everything was fixed. I'll check back through Grafana and see if I can find the patterns
.